src/components/CheckableInput/CheckableInput.tsx

```diff
@@ -3,20 +3,25 @@ import { nanoid } from "nanoid";
 import React, { useEffect, useRef, HTMLProps } from "react";
 import type { ReactNode } from "react";
 
-export type Props = {
-  /**
-   * The type of the input element.
-   */
-  inputType: "radio" | "checkbox";
-  /**
-   * The label for the input element.
-   */
-  label: ReactNode;
-  /**
-   * Whether the input element should display in indeterminate state.
-   */
-  indeterminate?: boolean;
-} & Omit<HTMLProps<HTMLInputElement>, "type">;
+import type { PropsWithSpread } from "types";
+
+export type Props = PropsWithSpread<
+  {
+    /**
+     * The type of the input element.
+     */
+    inputType: "radio" | "checkbox";
+    /**
+     * The label for the input element.
+     */
+    label: ReactNode;
+    /**
+     * Whether the input element should display in indeterminate state.
+     */
+    indeterminate?: boolean;
+  },
+  Omit<HTMLProps<HTMLInputElement>, "type">
```

Comment:
Does this still need to explicitly Omit type? If so we should maybe leave a comment as to why.

Reply:
Good call!